### PR TITLE
FoV layers Unit Test

### PR DESCRIPTION
**Pull request by Azarak**
## About The Pull Request
Originally I started doing a layer audit, but then I was made aware that one already existed. So here's this unit test I guess.
Will be adjusted to properly work with https://github.com/tgstation/tgstation/pull/63903 after it gets merged.

## Why It's Good For The Game
prevent bug

### CODE REVIEW

File: code/modules/unit_tests/mob_layers.dm [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Checks if the layers of all atoms are on the proper planes.
+/datum/unit_test/mob_layers/Run()
```
Comment (Mothblocks) [inline code/modules/unit_tests/mob_layers.dm:2]:
```suggestion
/// Checks if the layers of all atoms are on the proper planes.
/datum/unit_test/mob_layers

/datum/unit_test/mob_layers/Run()
```

Nicer for the autodoc site

File: code/modules/unit_tests/mob_layers.dm [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	for(var/atom/movable/am_path as anything in typesof(/atom/movable))
```
Comment (Mothblocks) [inline code/modules/unit_tests/mob_layers.dm:3]: Full name--`movable_path` or similar.

File: code/modules/unit_tests/mob_layers.dm [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+		switch(layer)
+			if(GAME_PLANE_FOV_HIDDEN_LAYER_START to GAME_PLANE_FOV_HIDDEN_LAYER_END)
+				if(plane != GAME_PLANE_FOV_HIDDEN)
+					Fail("[am_path] with layer [layer] should be in plane [GAME_PLANE_FOV_HIDDEN]. Currently is in [plane].")
+			if(ABOVE_GAME_PLANE_LAYER_START to ABOVE_GAME_PLANE_LAYER_END)
+				if(plane != ABOVE_GAME_PLANE)
+					Fail("[am_path] with layer [layer] should be in plane [ABOVE_GAME_PLANE]. Currently is in [plane].")
+			else
+				if(plane == ABOVE_GAME_PLANE || plane == GAME_PLANE_FOV_HIDDEN)
+					Fail("[am_path] with layer [layer] shouldn't be in either [ABOVE_GAME_PLANE] or [GAME_PLANE_FOV_HIDDEN] planes.")
```
Comment (Mothblocks) [inline code/modules/unit_tests/mob_layers.dm:17]: All of these should write out the word as well.

As in, "should be in plane [ABOVE_GAME_PLANE] (ABOVE_GAME_PLANE)".

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/unit_tests/_unit_tests.dm [hunk 4]
```diff
@@ -79,6 +79,7 @@
 #include "merge_type.dm"
 #include "metabolizing.dm"
 #include "mob_spawn.dm"
+#include "mob_layers.dm"
```
Comment (Mothblocks) [inline code/modules/unit_tests/_unit_tests.dm:82]: l comes before s

**Review by Mothblocks — APPROVED** [review]

**Comment by Azarak** [comment]
Dont know whether I should wait on https://github.com/tgstation/tgstation/pull/63903 to be merged or what, because I'd like to avoid conflicts if possible.

**Pull request closed without merging**